pkg/services/datasources/models.go

```diff
@@ -176,6 +177,9 @@ type AddDataSourceCommand struct {
 	ReadOnly                bool              `json:"-"`
 	EncryptedSecureJsonData map[string][]byte `json:"-"`
 	UpdateSecretFn          UpdateSecretFn    `json:"-"`
+
+	// Refactor to use User field instead of UserRequester
+	UserRequester identity.Requester
```

Comment:
Lets not add this field. We already have UserID so I think this could be handled where that property is set?

Reply:
What would you add to the service to get the user from the userID?

Reply:
https://github.com/grafana/grafana/pull/95072/commits/23cba787523dcff373a4b4d3d90d5f2ae923cd8b
Opted for using the userService

Reply:
@kalleep i opted instead of reverting back all changes for the datasource creation. As we can confirm that the current have tested it locally for service accounts getting Admin permissions.